app/controllers/api/v1/base_controller.rb

```diff
@@ -27,7 +27,7 @@ def log_exception exception
   end
 
   def authenticate_user_using_x_auth_token
-    user_email = params[:id].presence
+    user_email = params[:email].presence || request.headers['X-Auth-Email']
```

Comment:
@ershad _a Why should we support fetching the email from `params[:email]`? I think only headers would be a better way to go.